uaclient/clouds/identity.py

```diff
@@ -8,7 +8,7 @@
 CLOUDINIT_RESULT_FILE = "/var/lib/cloud/data/result.json"
 
 
-@util.retry(FileNotFoundError, [1, 2])
+@util.retry(FileNotFoundError, [1, 2, 3])
```

Comment:
Is this better handled by the service file? We could block on the cloud-id file?

Reply:
Good point, added and tested the following upstart change:

# Only start once cloud-init finishes detecting the DataSource on trusty
start on file FILE=/var/lib/cloud/data/result.json

Able to see that no retries were needed then, so I dropped the added retry.